dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -1311,6 +1313,20 @@ protected List<Node> convertCallArguments(
             } else {
                 assert lastParamType instanceof ArrayType
                         : "variable argument formal must be an array";
+                // Handle anonymous constructors that extend a class with an enclosing type.
+                // The logic is borrowed from AnnotatedTypes.adaptParameters(). However,
+                // adaptParameters
+                // does not need this part anymore after mering this PR.
+                if (method.getKind() == ElementKind.CONSTRUCTOR
+                        && method.getEnclosingElement().getSimpleName().contentEquals("")
+                        && enclosingType != null) {
+                    TypeMirror p0tm = formals.get(0);
+                    // We only care about Java 11+ since then the arguments do NOT include the
+                    // enclosing expression.
+                    if (types.isSameType(enclosingType, p0tm) && SystemUtil.jreVersion >= 11) {
```

Comment:
Do the simple int comparison first and then the type comparison.

Reply:
Thanks!